tests/e2e/specs/api-cache.test.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * API caching functions tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { visitAdminPage } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	resetSiteKit,
+	safeLoginUser,
+	setupSiteKit,
+	testSiteNotification,
+	useRequestInterception,
+	wpApiFetch,
+} from '../utils';
+import { deleteAuthCookie } from '../utils/delete-auth-cookie';
+
+const goToSiteKitDashboard = async () => {
+	await visitAdminPage( 'admin.php', 'page=googlesitekit-dashboard' );
+};
+
+describe( 'API cache', () => {
+	beforeAll( async () => {
+		await page.setRequestInterception( true );
+		useRequestInterception( ( request ) => {
+			const url = request.url();
+			if ( url.match( 'search-console/data/searchanalytics' ) ) {
+				request.respond( { status: 200, body: '[]' } );
+			} else if ( url.match( 'pagespeed-insights/data/pagespeed' ) ) {
+				request.respond( { status: 200, body: '{}' } );
+			} else if ( url.match( 'user/data/survey-timeouts' ) ) {
+				request.respond( { status: 200, body: '[]' } );
+			} else {
+				request.continue();
+			}
+		} );
+
+		await setupSiteKit();
+	} );
+
+	afterAll( async () => {
+		await resetSiteKit();
+	} );
+
+	it( 'isolates client storage between users', async () => {
```

Comment:
I just noticed the name here is slightly inaccurate since the cache is user-specific but also to the session which is what is actually being tested here.

```suggestion
	it( 'isolates client storage between sessions', async () => {
```